client/src/components/FlightSearch.js

```diff
@@ -1,123 +1,220 @@
+/** @format */
+
 import React, { useState } from "react";
 
-import { Grid, Paper, TextField, Button, InputLabel } from "@material-ui/core";
+import { Grid, Paper, TextField, Button } from "@material-ui/core";
 import { Autocomplete } from "@material-ui/lab";
+import DateFnsUtils from "@date-io/date-fns";
+import {
+	MuiPickersUtilsProvider,
+	KeyboardDatePicker,
+} from "@material-ui/pickers";
 
 import useStyles from "../styles/FlightSearch";
 
 // Mock Data
 const cities = [
-  { title: "Vancouver" },
-  { title: "Calgary" },
-  { title: "Toronto" },
-  { title: "Bangkok" },
+	{ title: "Vancouver", placeId: "" },
+	{ title: "Calgary", placeId: "" },
+	{ title: "Toronto", placeId: "" },
+	{ title: "Bangkok", placeId: "" },
 ];
 
-const FlightSearch = () => {
-  const classes = useStyles();
+const FlightSearch = ({ submit }) => {
+	const classes = useStyles();
 
-  const curr = new Date();
-  curr.setDate(curr.getDate());
-  const date = curr.toISOString().substr(0, 10);
+	const [from, setFrom] = useState("Vancouver");
+	const [to, setTo] = useState("Bangkok");
+	const [arrivalDate, setArrivalDate] = useState(new Date());
+	const [departDate, setDepartDate] = useState(new Date());
+	const [travellers, setTravellers] = useState(1);
+	const [dateError, setDateError] = useState(false);
+	const [fromError, setFromError] = useState(false);
+	const [toError, setToError] = useState(false);
 
-  const [from, setFrom] = useState("Vancouver");
-  const [to, setTo] = useState("Bangkok");
-  const [arrival, setArrival] = useState(date);
-  const [departure, setDeparture] = useState(date);
-  const [travellers, setTravellers] = useState(1);
+	const formatDate = (date) => {
+		const year = date.getFullYear();
+		const month = date.getMonth();
+		const day = date.getDate();
+		date = new Date(year, month, day);
+		return date.toISOString().split("T")[0];
+	};
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    const searchObject = {
-      from: from,
-      to: to,
-      arrival: arrival,
-      departure: departure,
-      travellers: travellers,
-    };
-  };
+	const handleFromLocation = (event, value) => {
+		setFrom(value);
+		setFromError(false);
+	};
+	const handleToLocation = (event, value) => {
+		setTo(value);
+		setToError(false);
+	};
+	const handleArrivalDate = (date) => setArrivalDate(date);
+	const handleDepartDate = (date) => {
+		setDepartDate(date);
+		setDateError(false);
+	};
+	const handleTravellers = (event) => setTravellers(event.target.value);
+	const handleSubmit = async (event) => {
+		event.preventDefault();
+		if (from && to && departDate >= arrivalDate) {
+			const fromCity = cities.find((city) => from === city.title).placeId;
+			const toCity = cities.find((city) => to === city.title).placeId;
```

Comment:
All your `placeId` are set to `""`.

Reply:
I am waiting on city data from Ray to add in the values.